### ISSUE TITLE
multiple column lambda function returns array, not scalar

### ISSUE THREAD

**Issue by davidwynter** [jmcarpenter2/swifter#85]
I cannot see why this returns an array. I use np.where in my function called by apply elsewhere in my code without issues.

    def calc_y_ratio(self, yield_0, yield_d):
        return np.where(round(yield_0, 8) != 0, yield_decay/abs(yield_0), np.nan)

    def calc_d(self, df):
        df['d_ratio'] = df[['y_0', 'y_d']].swifter.apply(
            lambda row: self.calc_y_ratio(row['y_0'], row['y_d']), axis=1)

        return df

    cols = ['y_0', 'y_d']

    df = pd.DataFrame(np.array([[0.5, -0.3],
                                    [0.5, 0.3],
                                    [-0.5, 0.3],
                                    [-0.5, -0.3],
                                    [0, -0.3]]), columns=cols)

    d_cols = [ 'd_ratio']
    cols += d_cols

    df_ref = pd.DataFrame(np.array([[0.5, -0.3, -0.6],
                                        [0.5, 0.3, 0.6],
                                        [-0.5, 0.3, 0.6],
                                        [-0.5, -0.3, -0.6],
                                        [0, -0.3, np.nan]]), columns=cols)

    df_test = df.copy()
    df_test = calc_d(df_test)

    pd.testing.assert_frame_equal(df_test, df_ref, check_exact=False, check_dtype=False)

Any ideas? Or have I missed something obvious?

**Comment by jmcarpenter2** [comment]
@davidwynter,

Thanks for raising this issue. I tested this code in my environment and the issue is not with `swifter`.

When you apply `calc_y_ratio` to the `df_test`, the values are correct, but the `dtype` of column `d_ratio` is changed to `object` for some reason. _This is also true for a regular pandas apply._ Conversely, when you instantiate the dataframe `df_ref`, it is creating column `d_ratio` as a `float64` type. Thus, when you `assert_frame_equal` using the pandas testing framework the result is a raised assertion of inequality.

[This stackoverflow post](https://stackoverflow.com/questions/31715287/data-type-of-pandas-column-changes-to-object-when-its-passed-to-a-function-via) seems relevant, though we don't appear to have mixed type here.

Long story short, this is an issue with pandas apply.

Thanks,
Jason

**Comment by jmcarpenter2** [comment]
Archiving due to inactivity..
